lib/fotofetch.rb

```diff
@@ -59,25 +81,27 @@ def height_ok?(sizes)
       (0 < height && height < sizes[1][1]) || height > (sizes[1][1] * 2)
     end
 
+    def root_url(link)
+      link.split("/")[2]
+    end
+
     # Adds root urls as hash keys
     def add_sources(urls)
-      results = {}.compare_by_identity
-      urls.each { |link| results[link.split("/")[2]] = link}
-      results
+      urls.map { |link| [root_url(link), link] }.to_h
```

Comment:
@plato721 thanks for the fix. And yep, definitely some things that can still be improved in there :)